### DISCUSSION TITLE
The story behind "FreshRSS"

### DISCUSSION THREAD

**Discussion by math-GH** [FreshRSS/FreshRSS#3840] [General]
Hey,

what is the story behind FreshRSS?

As I see it started in 2013.

What was the trigger to start the development of this cool application? Where there other apps, but they were not "good enough"?

And how it came to the name "FreshRSS"?

**Comment by Alkarex** [comment]
:-)
Although in French, we got a little interview some years ago: https://www.nextinpact.com/article/28150/106286-intelligence-social-freshrss-veut-redonner-coup-fouet-aux-agregateurs-flux
Can you understand any of it? (maybe with automatic translation)
I can happily provide more information about my take on it, both as text and through other channels, e.g. videoconference.
Ping @marienfressinaud

**Reply by Alkarex** [reply]
By the way, we should include a link to that article and maybe a couple more in our documentation somewhere

**Reply by math-GH** [reply]
Very cool article. Thanks to Google translate, that translates this article very well

**Reply by Frenzie** [reply]
@Alkarex I'm fairly sure it's already in there, unless you and/or @marienfressinaud happened to link it in chat.

**Reply by Alkarex** [reply]
> @Alkarex I'm fairly sure it's already in there, unless you and/or @marienfressinaud happened to link it in chat.

A quick search in our code / documentation base returns no result :-)

**Comment by marienfressinaud** [comment]
I’ll be happy to share a bit of history :)

First, you can find all the initial posts I wrote at the time on my personal blog (in French): https://marienfressinaud.fr/serie/freshrss.html

Here's the short story. The development of FreshRSS began in October 2012 ([first commit!](https://github.com/FreshRSS/FreshRSS/commit/fb57be5a5af3a2fb46b2dbf2b503ffe78eb5cf49)), but I wrote Minz (the framework) a bit before, [in 2011](https://marienfressinaud.fr/minz-is-not-zend.html) (in French). I was a student so I made a lot of mistakes and I would not recommend to use this framework… but hey, it (still) does the job :)

I was using RSSLounge at this time, which had too many bugs for me (can't remember the bugs exactly) and had too much of JavaScript. You can find a description and a screenshot [here](https://sumtips.com/web/host-personal-web-based-feed-reader/), but the developer abandoned the project and started [selfoss](https://selfoss.aditu.de/) later. [TinyTinyRSS](https://tt-rss.org/) was too complicated to install, as I remember, and quite heavy. Some other projects like [Leed](http://blog.idleman.fr/leed-lagregateur-rss-poid-plume/) (French) and [KrISS feed](http://tontof.net/kriss/feed/) were very young. Also, I had fun to start projects of my own, and I was intrigued of how RSS aggregators were working.

I started to write about FreshRSS almost from the beginning of the project and it quickly had a small audience thanks to sebsauvage, a well-known French free software blogger (not sure if I can describe him like that ^^). You can see [the first time](https://www.sebsauvage.net/links/index.php?amMBow) he mentioned FreshRSS was the day after I published my first post. It started to gain a lot more users when Google announced the end of Google Reader in 2013. It looks like @Alkarex joined me a bit after ([see its first contribution](https://github.com/FreshRSS/FreshRSS/commit/4cf9119a7dfd8b8a45344dff4d884a445dcb5a5a) :)).

Fun fact, my first user was my roommate. He told me the only thing preventing him to use FreshRSS was the interface… so I rewrote the entire theme which became the "Origine" theme (heavily inspired by Bootstrap, but I didn't want to include this library in FRSS).  I don't think I have any screenshots, but it could be fun to install this version :)

I stopped to contribute actively to FRSS in 2015, when I finished my studies. I  took a sabbatical year and I spent the first months to publish the version 1.0, announced on [the 31th January](https://marienfressinaud.fr/freshrss-un-point-zero.html). Then @Alkarex became the main maintainer.

I tried to come back in 2019 though: I started a SaaS version of FRSS at [flus.io](https://flus.io/i/), intended for a French audience, and I wanted to help on FRSS. I had a lot of ideas to improve the software, but I realized a few months later it would be very difficult to add them to FRSS. So I started [flusio](https://github.com/flusio/flusio), but that's a different story :) I am still considering to help more on FRSS, but it's quite difficult to get paid enough to live of my activity. So at the moment I'm focusing on what I think will generate the highest revenue ;)

That's being said, next year will be the 10th anniversary of FRSS and I want to dedicate a bit of my time to celebrate it :)

**Reply by marienfressinaud** [reply]
I forgot to explain the name! I can't remember exactly, but I think I chose it when I created the GitHub repository. Honestly, I think the name came quite randomly :sweat_smile: with the idea to convey a sense of novelty and being light. I like the feeling of "freshness", so I guess it was the same at the time. And "RSS" is pretty obvious :p

**Reply by math-GH** [reply]
That is cool. Thanks for sharing :)

**Comment by marienfressinaud** [comment]
Here we go! The very first version of FreshRSS:

![Screenshot 2021-09-11 at 20-41-59 Flux RSS](https://user-images.githubusercontent.com/1436309/132958162-69ec0fa2-31fc-4a83-bbfc-309ac78e5125.png)

**Comment by SolarDesalination** [comment]
Hi, why doesn't the FreshRSS website link to Flus for those that want to get a hosted option for FreshRSS?

**Reply by Alkarex** [reply]
Ping @marienfressinaud

**Reply by Alkarex** [reply]
See also this little list of options https://app.flus.fr/collections/1698850762083905588

**Reply by marienfressinaud** [reply]
In July, I will begin to work on a new website (see https://github.com/FreshRSS/FreshRSS/discussions/4294), it is intended to add a section with opened instances of FRSS. Something like https://joinmastodon.org/communities for Mastodon, but probably simpler.
